egui/src/widgets/plot/mod.rs

```diff
@@ -14,12 +14,14 @@ pub use items::{
 };
 pub use legend::{Corner, Legend};
 
+use self::items::HoverConfig;
+
 mod items;
 mod legend;
 mod transform;
 
-type CustomLabelFunc = dyn Fn(&str, &Value) -> String;
-type CustomLabelFuncRef = Option<Box<CustomLabelFunc>>;
+type CustomLabelFunc = dyn Fn(&HoverConfig, &str, &Value) -> String;
+type CustomLabelFuncRef = Box<CustomLabelFunc>;
```

Comment:
Since `CustomLabelFuncRef` seems to be the prevalent type used everywhere, maybe we should think about a shorter/more precise name? `BarChart` and `BoxPlot` have an `element_formatter()` method, which customizes the displayed text when hovering over them, which is essentially the same concept.

Maybe something like `HoverFormatter`?

On the other hand, the type `dyn Fn(&HoverConfig, &str, &Value) -> String` seems to be used only once, so it can either be inlined or use a more verbose name `HoverFormatterFn` or so.

We are going to need a lot of these customization functions, so agreeing on a term like "formatter" could make sense.
Use cases:
* Hovering for bar charts and box plots (currently `element_formatter`, could be renamed to `hover_formatter`)
* Hovering for X/Y values (this PR, maybe `hover_formatter` makes clear it's the same idea)
* Labeling X/Y axis (I'm going to work on this, I'm thinking about `x_axis_formatter` + `y_axis_formatter`)